numba/cuda/cudadrv/driver.py

```diff
@@ -1179,59 +1375,91 @@ def can_access_peer(self, peer_device):
         """Returns a bool indicating whether the peer access between the
         current and peer device is possible.
         """
-        can_access_peer = c_int()
-        driver.cuDeviceCanAccessPeer(
-            byref(can_access_peer),
-            self.device.id,
-            peer_device,
-        )
+        if config.CUDA_USE_CUDA_PYTHON:
+            peer_device = cuda_driver.CUdevice(peer_device)
+            can_access_peer = driver.cuDeviceCanAccessPeer(self.device.id,
+                                                           peer_device)
+        else:
+            can_access_peer = c_int()
+            driver.cuDeviceCanAccessPeer(byref(can_access_peer),
+                                         self.device.id, peer_device,)
+
         return bool(can_access_peer)
 
     def create_module_ptx(self, ptx):
         if isinstance(ptx, str):
             ptx = ptx.encode('utf8')
-        image = c_char_p(ptx)
+        if config.CUDA_USE_CUDA_PYTHON:
+            image = ptx
+        else:
+            image = c_char_p(ptx)
         return self.create_module_image(image)
 
     def create_module_image(self, image):
         module = load_module_image(self, image)
-        self.modules[module.handle.value] = module
+        if config.CUDA_USE_CUDA_PYTHON:
+            key = module.handle
+        else:
+            key = module.handle.value
+        self.modules[key] = module
         return weakref.proxy(module)
 
     def unload_module(self, module):
-        del self.modules[module.handle.value]
+        if config.CUDA_USE_CUDA_PYTHON:
+            key = module.handle
+        else:
+            key = module.handle.value
+        del self.modules[key]
 
     def get_default_stream(self):
-        handle = drvapi.cu_stream(drvapi.CU_STREAM_DEFAULT)
+        if config.CUDA_USE_CUDA_PYTHON:
+            handle = cuda_driver.CUstream(0)
```

Comment:
I'm trying to avoid using `drvapi` for the Nvidia bindings code, because that's for the ctypes binding and I'd like to try to avoid mixing them where possible. It's unfortunate that in the NV binding there is no definition of the default stream (there is a `CU_STREAM_DEFAULT`, but it is a stream creation flag and not a stream). I think I'll probably define the default stream for use with the Nvidia bindings in driver.py and use that where appropriate.

Reply:
Thanks, I think it makes sense to keep the two concerns separate, defining a default stream to avoid a magic number is a good workaround for now.